File: example/java/config/other/Loot.java
package config.other;

public class Loot {
    private int lootid;
    private String ename;
    private String name;
    private java.util.List<Integer> chanceList;
    private java.util.List<config.other.Lootitem> ListRefLootid;

    private Loot() {
    }

    public static Loot _create(configgen.genjava.ConfigInput input) {
        Loot self = new Loot();
        self.lootid = input.readInt();
        self.ename = input.readStr();
        self.name = input.readStr();
        {
            int c = input.readInt();
            if (c == 0) {
                self.chanceList = java.util.Collections.emptyList();
            } else {
                self.chanceList = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.chanceList.add(input.readInt());
                }
            }
        }
        return self;
    }

    /**
     * 序号
     */
    public int getLootid() {
        return lootid;
    }

    public String getEname() {
        return ename;
    }

    /**
     * 名字
     */
    public String getName() {
        return name;
    }

    /**
     * 掉落0件物品的概率
     */
    public java.util.List<Integer> getChanceList() {
        return chanceList;
    }

    public java.util.List<config.other.Lootitem> listRefLootid() {
        return ListRefLootid;
    }

    @Override
    public String toString() {
        return "(" + lootid + "," + ename + "," + name + "," + chanceList + ")";
    }

    public void _resolveDirect(config.ConfigMgr mgr) {
        ListRefLootid = new java.util.ArrayList<>();
        for (config.other.Lootitem v : mgr.other_lootitem_All.values()) {
            if (v.getLootid() == lootid)
                ListRefLootid.add(v);
        }
        ListRefLootid = ListRefLootid.isEmpty() ? java.util.Collections.emptyList() : new java.util.ArrayList<>(ListRefLootid);
    }

    public void _resolve(config.ConfigMgr mgr) {
        _resolveDirect(mgr);
    }

    public static Loot get(int lootid) {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.getOtherLoot(lootid);
    }

    public static java.util.Collection<Loot> all() {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.allOtherLoot();
    }

    public static class _ConfigLoader implements config.ConfigLoader {

        @Override
        public void createAll(config.ConfigMgr mgr, configgen.genjava.ConfigInput input) {
            for (int c = input.readInt(); c > 0; c--) {
                Loot self = Loot._create(input);
                mgr.other_loot_All.put(self.lootid, self);
            }
        }

        @Override
        public void resolveAll(config.ConfigMgr mgr) {
            for (Loot e : mgr.other_loot_All.values()) {
                e._resolve(mgr);
            }
        }

    }

}

File: example/go/config/Other/Loot.go
package other

type Loot struct {
    lootid int //序号
    ename string
    name string //名字
    chanceList []int //掉落0件物品的概率
    listRefLootid List<config.Other.Lootitem>
}

//getters
func (t *Loot) GetLootid() int {
    return t.lootid
}

func (t *Loot) GetEname() string {
    return t.ename
}

func (t *Loot) GetName() string {
    return t.name
}

func (t *Loot) GetChanceList() []int {
    return t.chanceList
}

//ref properties
func (t *other.loot) GetListRefLootid() List<config.Other.Lootitem> {
    return t.listRefLootid
}

var all []Loot
func GetAll() []Loot {
    return all[:len(all)]:len(all)]
}

var allMap map[int]Loot
func Get(key int) (Loot,bool){
    return allMap[key]
}
        public static Loot Get(int lootid)
        {
            Loot v;
            return all.TryGetValue(lootid, out v) ? v : null;
        }

        public static List<Loot> All()
        {
            return all.OrderedValues;
        }

        public static List<Loot> Filter(Predicate<Loot> predicate)
        {
            var r = new List<Loot>();
            foreach (var e in all.OrderedValues)
            {
                if (predicate(e))
                    r.Add(e);
            }
            return r;
        }

        internal static void Initialize(Config.Stream os, Config.LoadErrors errors)
        {
            all = new Config.KeyedList<int, Loot>();
            for (var c = os.ReadInt32(); c > 0; c--) {
                var self = _create(os);
                all.Add(self.Lootid, self);
            }
        }

        internal static void Resolve(Config.LoadErrors errors) {
            foreach (var v in All())
                v._resolve(errors);
        }

